akka-actor/src/main/scala/akka/actor/ActorSystemSetup.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor
+
+import java.util.Optional
+
+import scala.annotation.varargs
+import scala.compat.java8.OptionConverters._
+import scala.reflect.ClassTag
+
+/**
+ * Marker supertype for a setup part that can be put inside [[ActorSystemSetup]], if a specific concrete setup
+ * is not specified in the actor system setup that means defaults are used (usually from the config file) - no concrete
+ * setup instance should be mandatory in the [[ActorSystemSetup]] that an actor system is created with.
+ */
+abstract class Setup {
+
+  /**
+   * Construct an [[ActorSystemSetup]] with this setup combined with another one. Allows for
+   * fluent creation of settings.
+   */
+  final def and(other: Setup): ActorSystemSetup = ActorSystemSetup(this, other)
```

Comment:
We had that discussion, it was chosen because that is what it is called in `Attributes` where it is also not commutative. I don't think that aspect of it will be confusing it is only here to be able to do:
```
ActorSystem("name", bootstrapSetup and remotingSettings and clusterSettings)
```